tv-show/src/components/features/shows/ShowReviewSection/ShowReviewSection.tsx

```diff
@@ -0,0 +1,82 @@
+import { IReview } from "@/typings/Review.type";
+import { Box, Card, CardBody, Container, Heading, Stack } from "@chakra-ui/react";
+import ReviewList from "../../review/ReviewList/ReviewList";
+import ReviewForm from "../../review/ReviewForm/ReviewForm";
+import { useEffect, useState } from "react";
+
+const LOCAL_STORAGE_KEY = 'reviews';
+
+function loadReviewsFromLocalStorage(): IReview[] {
+  const reviewsString = localStorage.getItem(LOCAL_STORAGE_KEY);
+  if(!reviewsString) {
+    return [];
+  }
+  return JSON.parse(reviewsString);
+}
+
+function storeReviewsToLocalStorage(reviews: IReview[]) {
+  localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(reviews));
+}
+
+function calculateAverageRating(reviews: IReview[]) {
+  let averageRating = 0;
+
+  reviews.forEach((review) => averageRating += review.rating);
+  if(reviews.length) {
+    averageRating /= reviews.length;
+  }
+
+  return averageRating ? averageRating : undefined;
+}
+
+interface IShowReviewSection {
+  setAverageRating: (avg: number | undefined) => void;
+}
+
+export default function ShowReviewSection({setAverageRating}: IShowReviewSection) {
+  const [reviews, setReviews] = useState<IReview[]>([]);
+  const [updated, setUpdated] = useState(false);
+  const onSubmitClick = (newReview: IReview) => {
+    setReviews([...reviews, newReview]);
+    setUpdated(true);
+  };
+  const onRemoveClick = (deletedReview: IReview) => {
+    const updatedReviews = reviews.filter((review) => review != deletedReview);
+    setReviews(updatedReviews);
+    setUpdated(true);
+  };
+
+  useEffect(() => {
+    const localStorageReviews = loadReviewsFromLocalStorage();
+    setReviews(localStorageReviews);
+  }, []);
+
+  useEffect(() => {
+    const averageRating = calculateAverageRating(reviews);
+    setAverageRating(averageRating);
+    if(updated) {
```

Comment:
What are you trying to achieve with this 'updated' state?

Reply:
The data was being overwritten, because `reviews` are updated by setting the state and by reading the mock values, so I ended up with an empty array on each reload.
To reference the next comment, `updated` shouldn't be updated without the reviews also being updated. And by setting `updated` to false after storing, I would be retriggering the average calculation unnecessarily?